planner/plansdk/plansdk.go

```diff
@@ -16,12 +16,24 @@ type PlanError struct {
 	error
 }
 
+// TODO: Plan currently has a bunch of fields that it should not export.
+// Two reasons why we need this right now:
+// 1/ So that individual planners can use the fields
+// 2/ So that we print them out correctly in `devbox plan`
+//
+// (1) can be solved by using a WithOption pattern, (e.g. NewPlan(..., WithWelcomeMessage(...)))
+// (2) can be solved by using a custom JSON marshaler.
 type Plan struct {
```

Comment:
Let's put an empty line between this comment and Plan so it doesn't show up in godocs.